src/context/ExecutionContextManager.ts

```diff
@@ -21,6 +22,16 @@ export default class ExecutionContextManager {
      * @param provided the {@link ExecutionContext} provider to be set
      */
     static setProvider(provider: any) {
+
+        const contextCanChangeable = contextProvider.canChangeablebleContext();
+        if (!contextCanChangeable) {
+
+            ThundraLogger.debug('<ExecutionContextManager> Default context provider set as ForesightContextProvider.'
```

Comment:
I think it might be better to write `contextProvider` name (maybe we should add name to context provider) here instead of using `ForesightContextProvider` as constant as in theory it might be another non-changable context